libraries/CurieBLE/src/internal/BLECharacteristicImp.h

```diff
@@ -324,7 +324,7 @@ class BLECharacteristicImp: public BLEAttribute{
     bt_gatt_subscribe_params_t _sub_params;
     bool        _subscribed;
     
-    bool _reading;
+    volatile bool _reading;
```

Comment:
When read response come back, the setValue will be called and clear the reading flag. Because the response is asynchronous. So the below code may optimized for the value was set true before call while.

_reading = true;

while(_reading)
{
......
}